kafka/src/main/java/com/metamx/tranquility/kafka/model/TranquilityConfig.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.metamx.tranquility.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang.Validate;
+import org.joda.time.Period;
+
+/**
+ * Configuration options for Tranquility.
+ */
+public class TranquilityConfig
+{
+  @JsonProperty
+  private int batchSize = 5000;
+
+  @JsonProperty
+  private int maxPendingBatches = 100;
+
+  @JsonProperty
+  private Period windowPeriod = new Period("PT10M");
+
+  @JsonProperty
+  private int partitions = 1;
+
+  @JsonProperty
+  private int replicants = 1;
+
+  @JsonProperty
+  private boolean useTopicAsDataSource = false;
+
+  public int getBatchSize()
+  {
+    return batchSize;
+  }
+
+  public void setBatchSize(int batchSize)
```

Comment:
If you want Jackson to use these validations rather than tweak the fields with reflection, I think you need to have the `@JsonProperty` here rather than on the field.

Although, if you don't need to be able to set things after construction, you could also use a `@JsonCreator` constructor and not have any setters. That is the standard way that we do immutable Jackson-enabled objects in java.